include/common/arch/mips/asm/regdef.h

```diff
@@ -0,0 +1,41 @@
+#ifndef __ASM_REGDEF_H
+#define __ASM_REGDEF_H
+
```

Comment:
The register definitions are repeated at least 3 places. Can you please use a single include file everywhere?